source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -559,7 +559,16 @@ void Filter::onReceiveMessage(std::unique_ptr<ProcessingResponse>&& r) {
 
   // Check whether the server is asking to extend the timer.
   if (response->has_override_message_timeout()) {
-    onNewTimeout(DurationUtil::durationToMilliseconds(response->override_message_timeout()));
+    // The override_message_timeout in response may be too big for duration, which leads
+    // to exception been thrown during durationToMilliseconds() check.
+    // Needs to properly handle this case.
+    try {
```

Comment:
Adding PGV in ProcessingResponse proto for the override_message_timeout field does not help since the response message is handled by Envoy dataplane.  Has to handle the exception explicitly here.  Without this "try ... catch...", a large override_message_timeout will cause Envoy throw exception and crash due to this check: 

https://github.com/envoyproxy/envoy/blob/b896b6dfc266e6a16f7106da2f17f437db1d7f5c/source/common/protobuf/utility.cc#L654

Reply:
It may be better to keep a consistent limit for all the override message timeout fields. If so, need to ignore the field here also when it's value is larger than 3600, and the field's comments should be updated to reflect the limit.

Reply:
Thanks for the comments.   

Let me add some background for this crash:

Unlike the two timer related fields in ext_proc filter config above, this field : override_message_timeout: https://github.com/envoyproxy/envoy/blob/eee6b6d48d41ac9413c50049106d14537a65d7ad/api/envoy/service/ext_proc/v3/external_processor.proto#L188

 is sent back by the ext_proc service, and handled by Envoy dataplane.  It's upper bounded by the ext_proc filter config: https://github.com/envoyproxy/envoy/blob/d93958dbef1c2b11587bea6f3eb561bd630b212d/api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto#L170

This upper bound is enforced by the ext_proc filter code: 
https://github.com/envoyproxy/envoy/blob/d93958dbef1c2b11587bea6f3eb561bd630b212d/source/extensions/filters/http/ext_proc/ext_proc.cc#L534

However, ext_proc service can send any random number in this override_message_timeout field in its response. Before the code hit this upper bound check, it first call this function: DurationUtil::durationToMilliseconds(response->override_message_timeout()) to convert the Duration into milliseconds:  
https://github.com/envoyproxy/envoy/blob/d93958dbef1c2b11587bea6f3eb561bd630b212d/source/extensions/filters/http/ext_proc/ext_proc.cc#L562

In which, it validates the number, and if the number is out-of-range, it throw an exception:
https://github.com/envoyproxy/envoy/blob/b896b6dfc266e6a16f7106da2f17f437db1d7f5c/source/common/protobuf/utility.cc#L654

The response is handled by Envoy dataplane, thus adding the PGVs for this override_message_timeout field does not make any difference. i.e, even with PGVs for this proto field, ext_proc service can still send a random large number, the code still get into DurationUtil::durationToMilliseconds(response->override_message_timeout()) call and throw an exception there.  Thus, the ext_proc filter code has to handle this exception there and properly handles it.

Reply:
Yeah, what you did is good. My suggestion was about the maximal value validation.
IIUC what you wrote, this is where it is being validated:
https://github.com/envoyproxy/envoy/blob/d93958dbef1c2b11587bea6f3eb561bd630b212d/source/extensions/filters/http/ext_proc/ext_proc.cc#L534 

thanks

Reply:
Yeah, as discussed offline, I changed the code a little bit by passing in a "duration" parameter to the onNewTimeout() thus the out-of-range check is also done in that function.